object_templates/fact_spec.erb

```diff
@@ -7,9 +7,42 @@ require 'facter/<%= name %>'
 describe <%= name.to_sym.inspect %>, type: :fact do
   subject(:fact) { Facter.fact(<%= name.to_sym.inspect %>) }
 
+  # Mocking Facts
+  # You will most likely need to mock other facts if your custom fact relies up on them
+  # You can mock a existing core fact via:
+  #  allow(Facter.fact(:fqdn)).to receive(:value).and_return('test.example.com')
+  #  allow(Facter).to receive(:value).with(:fqdn).and_return('test.example.com')
+  # This all depends on how you utilize it in the fact code
+
+  #
+  # If you need to Mock a custom fact or non core fact you will first need to add that fact
+  # by requiring the facter file ie. require facter/ec2_metadata
+  # Or via code in the before each block:
+  # before :each do
+  #   Facter.add(:ec2_metadata) {}
+  # Once the custom fact is added, you can mock like
+  # allow(Facter.fact(:ec2_metadata)).to receive(:value).and_return({'42'})
+  # allow(Facter).to receive(:value).with(:fqdn).and_return('test.example.com')
+  # This all depends on how you utilize it in the fact code
+  # This mock will go inside your test block or the before each block
+
+  # Mocking confine example
+  # confine kernel: 'Linux' (Located in Fact)
+  # allow(Facter.fact(:kernel)).to receive(:value).and_return('Linux')
+
+  # Using and Mocking Exec
+  # You should always use the Facter Exectution method to execute commands on a system
+  # This will allow you to easily mock items as a result
+  #
```

Comment:
Same goes here. Maybe just have a new line and remove the comment.

Reply:
updated